<h2>Description</h2> 
<p> This track lists amino acid positions that are close to 
certain Covid drug target protein binding sites; potentially relevant 
for viral drug resistance. The protein products are the S, NSP3, and NSP5 proteins. The data is 
derived from 3D Protein Data Bank (PDB) X-ray crystal structures and was imported from 
<a target=_blank href="https://coronavirus3d.org/#/drug">
Coronavirus3d.org</a>, a project funded by NIAID, NIGMS and UCR. For more information, please 
visit that site and look through the 9 protein-drug structures listed.
</p>

<h2>Display Conventions</h2>
The display shows amino acids within 5 Angstroms of an inhibitor binding site on the
S, NSP3, and NSP5 proteins. The label of the annotations 
is the number of GISAID sequences with at least one mutation
at this position as of March 3, 2022. Click or mouse-over 
the mutations to show the full list of nucleotide changes and the 
frequency of each.

<h2>Data Access</h2>
<p>
You can see the original data
as tables on <a target=_blank href="https://coronavirus3d.org/#/drug">Coronavirus3d.org</a>.
The data can be explored interactively at the Genome Browser with the
<a href="hgTables">Table Browser</a> or the
<a href="hgIntegrator">Data Integrator</a>.
For automated analysis, the genome annotation is stored in
a bigBed file that can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/wuhCor1/">the download server</a>
or the <a href="../../goldenPath/help/api.html">API</a>.
</p>
<p>
Please refer to our
<a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">
mailing list archives</a> for questions, or our
<a href="../../FAQ/FAQdownloads.html#download36">Data Access FAQ</a>
for more information.</p>

<h2>Credits</h2>
<p>
Thanks to <a target="_blank" href="https://www.bv-brc.org/team">Anna Niewiadomska from 
the BV-BRC team</a> at the J. Craig Venter Institute for finding the dataset and testing
the track. Thanks to the Godzik Lab for making available the data in .tsv format
on the coronavirus3d.org website.</p>

