<H2>Description</H2>
<P>
This track shows regions of the human genome that are alignable
to other Homo sapiens genomes.
The alignable parts are shown with thick blocks that look like exons.
Non-alignable parts between these are shown with thin lines like introns.
More description on this display can be found below.
</P>

<P>
Other assemblies included in this track are from the HPRC project.:
</P>

<H3>Chain Track</H3>
<P>
The chain track shows alignments of the human genome to other
Homo sapiens genomes using a gap scoring system that allows longer gaps
than traditional affine gap scoring systems. It can also tolerate gaps in both
source and target assemblies simultaneously. These
&quot;double-sided&quot; gaps can be caused by local inversions and
overlapping deletions in both species.
<P>
The chain track displays boxes joined together by either single or
double lines. The boxes represent aligning regions.
Single lines indicate gaps that are largely due to a deletion in the
query assembly or an insertion in the target assembly.
assembly.  Double lines represent more complex gaps that involve substantial
sequence in both species. This may result from inversions, overlapping
deletions, an abundance of local mutation, or an unsequenced gap in one
species.  In cases where multiple chains align over a particular region of
the target genome, the chains with single-lined gaps are often
due to processed pseudogenes, while chains with double-lined gaps are more
often due to paralogs and unprocessed pseudogenes.</P>
<P>
In the &quot;pack&quot; and &quot;full&quot; display
modes, the individual feature names indicate the chromosome, strand, and
location (in thousands) of the match for each matching alignment.</P>

<H2>Display Conventions and Configuration</H2>
<H3>Chain Track</H3>
<P>By default, the chains to chromosome-based assemblies are colored
based on which chromosome they map to in the aligning organism. To turn
off the coloring, check the &quot;off&quot; button next to: Color
track based on chromosome.</P>
<P>
To display only the chains of one chromosome in the aligning
organism, enter the name of that chromosome (e.g. chr4) in box next to:
Filter by chromosome.</P>

<H2>Methods</H2>
<H1>TBD: This description needs to tell the story of how HPRC did these alignments.</H1>

<H2>Credits</H2>
<H1>TBD: need credits here of who made this data</H1>

<h2>References</h2>
<H1>TBD: references to HPRC resources need to be here</H1>
