<h2>Description</h2>

<p>
This track shows short nucleotide variants of a few basepairs when aligning
HPRC genomes to the hg38 reference assembly. The alignment was made with the
Minigraph-cactus approach described in the references below.
</p>

<p>There are three subtracks in this composite container track:
<ol>
    <li>All short variants &lt;= 3 bp long
    <li>All short variants &gt; 3 bp long
    <li>All short variants up to 50bp, without any length filter
</ol>

<h2>Methods</h2>
<p>
The Minigraph-Cactus HPRC v1.0 graph was converted to VCF using vg deconstruct.
This result was further postprocessed using vcfbub to flatten nested sites then
vcfwave to normalize by realigning alt alleles to the reference.  All steps are
described in Hickey et al 2023.  The postprocessing command lines and data can be found on 
<a href="https://github.com/human-pangenomics/hpp_pangenome_resources/blob/main/hprc-v1.0-mc.md#vcf-decomposition">Github</a>.
Finally, the resulting VCF was filtered by length and split into two VCFs using a cutoff of 3bp.
</p>

<h2>Credits</h2>
<p>
Thanks to Glenn Hickey for providing the HAL file from the HPRC project and for making these VCFs from them.
</p>

<h2>References</h2>

<p>
Wen-Wei Liao, Mobin Asri, Jana Ebler, ...<em>et al</em>, Heng Lin,
Benedict Paten
<a href='https://pubmed.ncbi.nlm.nih.gov/37165242/' target='_blank'>
A draft human pangenome reference</a>.
<em>Nature</em>. 2023 May;617(7960):312-324.
PMID: <a href='https://pubmed.ncbi.nlm.nih.gov/37165242/' target='_blank'>37165242</a>;
PMC: <a href='http://www.ncbi.nlm.nih.gov/pmc/articles/pmc10172123/' target=_blank>PMC1017212</a>;
DOI: <a href='https://doi.org/10.1038/s41586-023-05896-x' target='_blank'>10.1038/s41586-023-05896-x</a>
</p>

<p>
Glenn Hickey, Jean Monlong, Jana Ebler, Adam M Novak, Jordan M Eizenga,
Yan Gao; Human Pangenome Reference Consortium; Tobias Marschall, Heng Li,
Benedict Paten
<a href='https://pubmed.ncbi.nlm.nih.gov/37165083/' target='_blank'>
Pangenome graph construction from genome alignments with Minigraph-Cactus</a>.
<em>Nature Biotechnology</em>. 2023 May 10. doi: 10.1038/s41587-023-01793-w.
PMID: <a href='https://pubmed.ncbi.nlm.nih.gov/37165083/' target='_blank'>37165083</a>;
DOI: <a href=;https://doi.org/10.1038/s41587-023-01793-w; target='_blank'>10.1038/s41587-023-01793-w</a>
</p>

<p>
Armstrong J, Hickey G, Diekhans M, Fiddes IT, Novak AM, Deran A, Fang Q,
Xie D, Feng S, Stiller J
<em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/33177663" target="_blank">
Progressive Cactus is a multiple-genome aligner for the thousand-genome era</a>.
<em>Nature</em>. 2020 Nov;587(7833):246-251.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33177663" target="_blank">33177663</a>;
PMC: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7673649/" target="_blank">PMC7673649</a>;
DOI: <a href="https://doi.org/10.1038/s41586-020-2871-y" target="_blank">10.1038/s41586-020-2871-y</a>
</p>

<p>
Paten B, Earl D, Nguyen N, Diekhans M, Zerbino D, Haussler D.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/21665927" target="_blank">
Cactus: Algorithms for genome multiple sequence alignment</a>.
<em>Genome Res</em>. 2011 Sep;21(9):1512-28.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21665927" target="_blank">21665927</a>;
PMC: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3166836/" target="_blank">PMC3166836</a>;
DOI: <a href="https://doi.org/10.1101/gr.123356.111" target="_blank">10.1101/gr.123356.111</a>
</p>

