<!DOCTYPE html>
<html>
<head>
</head>

<body>
<h2>Description</h2>
<p>
SpliceAI is an <a href="https://github.com/Illumina/SpliceAI" target="_blank">open-source</a> deep
learning splicing prediction algorithm that can predict splicing alterations caused by DNA variations. 
Such variants may activate nearby cryptic splice sites, leading to abnormal transcript isoforms.
SpliceAI was developed at Illumina; a 
<a href="https://spliceailookup.broadinstitute.org" target="_blank">lookup tool</a> 
is provided by the Broad institute.
</p>

<h2>Display Conventions</h2>
<p>
Variants are colored by their predicted effects:
<ul>
<li><b><font color="#FF0000">Acceptor gain (red)</font></b> </li>
<li><b><font color="#FF8000">Acceptor loss (orange)</font></b> </li>
<li><b><font color="#0000FF">Donor gain (blue)</font></b> </li>
<li><b><font color="#D400FF">Donor loss (violet)</font></b> </li>
</ul>
Mouseover on items shows the variant, gene name, type of change (donor gain/loss, acceptor gain/loss), 
location of affected cryptic splice, and spliceAI score. Clicking on any item brings up a table with this
information.
</p>

<h2>Data Access</h2>
The raw data can be explored interactively with the
<a href="https://genome.ucsc.edu/cgi-bin/hgTables">Table Browser</a> or the
<a href="https://genome.ucsc.edu/cgi-bin/hgIntegrator">Data Integrator</a>.
For automated analysis, the data may be queried from our
<a href="https://genome.ucsc.edu/goldenPath/help/api.html">REST API</a>.<br>
Please refer to our
<a href="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">mailing list archives</a>
for questions, or our
<a href="https://genome.ucsc.edu/FAQ/FAQdownloads.html#downloads36">Data Access FAQ</a>
for more information.
<p>
SpliceAI scores for hg38 were obtained in VCF format from 
<a href="https://ftp.ensembl.org/pub/data_files/homo_sapiens/GRCh38/variation_plugins" target="_blank">the Ensembl ftp site</a>.
</p>

<h2>Methods</h2>
<p>
The spliceAI scores are represented in the VCF INFO field as 
<code style="background-color: lightgray;">SpliceAI=G|OR4F5|0.01|0.00|0.00|0.00|-32|49|-40|-31</code> <br>
Here, the pipe-separated fields contain 
<ul>
<li>ALT allele</li>
<li>Gene name</li>
<li>Acceptor gain score</li>
<li>Acceptor loss score</li>
<li>Donor gain score</li>
<li>Donor loss score</li>
<li>Relative location of affected cryptic acceptor</li>
<li>Relative location of affected acceptor</li>
<li>Relative location of affected cryptic donor</li>
<li>Relative location of affected donor</li>
</ul>
To create the track we selected only those variants with a score equal to or greater than 0.02.
</p>

<h2>References</h2>
<p>
Jaganathan K, Kyriazopoulou Panagiotopoulou S, McRae JF, Darbandi SF, Knowles D, Li YI, Kosmicki JA,
Arbelaez J, Cui W, Schwartz GB <em>et al</em>.
<a href="https://linkinghub.elsevier.com/retrieve/pii/S0092-8674(18)31629-5" target="_blank">
Predicting Splicing from Primary Sequence with Deep Learning</a>.
<em>Cell</em>. 2019 Jan 24;176(3):535-548.e24.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30661751" target="_blank">30661751</a>
</p>
</body>
</html>
