<!DOCTYPE html>
<!--#set var="TITLE" value="UCSC Genome Browser: Citing Us" -->
<!--#set var="ROOT" value="." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Citing the UCSC Browser in a Publication or Web Page</h1>
<p>
You are free to use all graphics produced by this website in your published work or on other websites, there
is no need to contact us to request permission, see the <a href="/license/">license page</a> for details.
But the UCSC Genome Browser is exclusively funded by NIH and if research articles do 
not cite or mention us, we cannot report the citations and it is harder for NIH to justify funding one of the 
few long-term and stable resources in genomics.
</p>

<p>
So you can do us a great favor if you cite the most recent NAR paper, the one
at the top of this <a
href="https://pubmed.ncbi.nlm.nih.gov/?term=%28%28UCSC+Genome+Browser%5BTitle%5D%29+AND+%28update%5BTitle%5D+OR+updates%5BTitle%5D%29+AND+%22Nucleic+acids+research%22%5BJournal%5D%29+NOT+ENCODE%5BTitle%5D&sort=date&size=100>"PubMed
search.</a>.  At the moment, the most recent paper has the PMID 36420891 and
the DOI 10.1093/nar/gkac1072.  In addition, you can include a reference to the Genome 
Browser website in your manuscript: <a href="index.html" 
target="_blank">http://genome.ucsc.edu</a> or mention in the text that you used the "UCSC Genome Browser". 
To cite just one of our computational tools, see the <a href="#ref">reference section</a> below.
</p>

<p>While doing this, you can check if the genome coordinates in your paper are
clear to the readers and if you specified the genome assembly that you used.
One cannot mention the genome assembly too often. Computational
biologists will appreciate if your supplemental files with genome annotations
include the genome assembly identifier directly in the file name. We try to
keep the identifiers short, currently they are "hg19" and "hg38". In a similar
vein, references to nucleotides ideally have the full chromosome position in
your publication text somewhere or in a supplemental table, and not only the
position on the transcript and or coding sequence, as transcript models can
change a lot.
</p>

<p>To make this even easier, when refering to a gene or genomic location, you can 
use the menu <a href="../cgi-bin/hgSession">"My Data &gt; My Sessions"</a>, to
generate a short, stable link to your current genome browser view for your
publication. In this way, readers can jump from your publication to this
genome location and get their view configured for your particular message.
You can even add your own annotations to the UCSC Genome Browser, either as
very simple text labels, by pasting a line like "chr1 1 1000 my_annotation"
into the box found under <a href="../cgi-bin/hgCustom">"My Data &gt; Custom
Tracks"</a> or by loading all genomics results into the browser via Track Hubs.
</p>

<a name="ref"></a>
<p>
<h2>General Genome Browser software and website references</h2>
<p>
<strong>Original UCSC Genome Browser publication:</strong>
Kent WJ, Sugnet CW, Furey TS, Roskin KM, Pringle TH, Zahler AM, Haussler D.  
<a href="http://www.genome.org/cgi/content/abstract/12/6/996" target="_blank">The human genome 
browser at UCSC</a>. 
<em>Genome Res.</em> 2002 Jun;12(6):996-1006.</p>

<p>
<strong>Genome Browser screenshots:</strong>
Journal-quality screenshots of the Genome Browser can be produced by using the View -&gt;PDF
utility in the top navigation bar from the tracks page. When including a
screenshot in your manuscript, reference <em>http://genome.ucsc.edu</em> in the caption and cite
the most recent Genome Browser paper in the manuscript. As mentioned above, we recommend including a <a 
href="goldenPath/help/hgSessionHelp.html#Introduction">My Data &gt; My Sesssion URL</a>
of your screenshot to allow readers to interact with your data and get more information.</p>
<p>
<strong>BLAT:</strong> 
Kent WJ. 
<a href="http://www.genome.org/cgi/content/abstract/12/4/656" target="_blank">BLAT - the BLAST-like 
alignment tool</a>. 
<em>Genome Res.</em> 2002 Apr;12(4):656-64.</p>

<p>
<strong>UCSC Table Browser:</strong>
Karolchik D, Hinrichs AS, Furey TS, Roskin KM, Sugnet CW, Haussler D, Kent WJ. 
<a href="http://nar.oupjournals.org/cgi/content/abstract/32/suppl_1/D493?ijkey=06tIQcBr2VZNz&keytype=ref" 
target="_blank">The UCSC Table Browser data retrieval tool</a>. 
<em>Nucleic Acids Res</em>. 2004 Jan 1;32(Database issue):D493-6.</p>
<p>
<strong>BigWig and BigBed tools:</strong>
Kent WJ, Zweig AS, Barber G, Hinrichs AS, Karolchik D.  
<a href="http://bioinformatics.oxfordjournals.org/content/26/17/2204.long" target="_blank">BigWig 
and BigBed: enabling browsing of large distributed data sets</a>.
<em>Bioinformatics</em>. 2010 Sep 1;26(17):2204-7.</p>
<p>
<strong>Track Data Hubs:</strong>
Raney BJ, Dreszer TR, Barber GP, Clawson H, Fujita PA, Wang T, Nguyen N, Paten B, Zweig AS, 
Karolchik D, Kent WJ.
<a href="http://bioinformatics.oxfordjournals.org/content/30/7/1003"
target="_blank">Track Data Hubs enable visualization of user-defined genome-wide annotations on the 
UCSC Genome Browser</a>.
<em>Bioinformatics</em>. 2014 Apr 1;30(7):1003-5. Epub 2013 Nov 13.</p>
<p>
<strong>UCSC Gene Sorter:</strong>
Kent WJ, Hsu F, Karolchik D, Kuhn RM, Clawson H, Trumbower H, Haussler D. 
<a href="http://www.genome.org/cgi/content/abstract/15/5/737" target="_blank">Exploring 
relationships and mining data with the UCSC Gene Sorter</a>.
<em>Genome Res</em>. 2005 May;15(5):737-41.</p>
<p>
<strong>UCSC SARS-CoV-2 Genome Browser</strong>
Fernandes JD, Hinrichs AS, Clawson H, et al. 
<a href="https://www.nature.com/articles/s41588-020-0700-8" 
target="_blank">The UCSC SARS-CoV-2 Genome Browser</a>. 
<em>Nat Genet.</em> 2020 Sept 9; 52:991-998.</p>

<strong>Other genome assemblies:</strong> 
Manuscripts that include the use of Genome Browser assembly data from organisms other than human 
should, e.g. one of the thousands of GenArk assemblies that we provide, should cite the relevant sequencing paper. 
Usually this information can be found on the 
assembly description that appears when the genome is selected on our gateway page or on the NCBI GenBank BioProject
page of the assembly.</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
